itests/src/test/java/fi/metatavu/edelphi/test/mock/QueryMocker.java

```diff
@@ -0,0 +1,272 @@
+
+      if (settingTypes != null && settingTypes.containsKey(settingKey)) {
+        settingType = settingTypes.get(settingKey);
+      }
+
+      if (settingType == SettingType.QUERY_PAGE_IDS_BY_TYPE) {
+        settingValue = StringUtils.join(getPageIdsByType(settingValue), ",");
+      }
+
+      createQueryPageSetting(id, settingKey, settingValue);
+    }
+
+    return id;
+  }
+
+  private long createQueryPageSetting(Long queryPageId, String key, String value) {
+    long id = getNextId("QueryPageSetting");
+    String sql = "INSERT INTO " + "  QueryPageSetting (id, key_id, queryPage_id, value) " + "VALUES "
+        + "  (?, (select id from QueryPageSettingKey where name = ?), ?, ?)";
+
+    executeSql(sql, id, key, queryPageId, value);
+
+    return id;
+  }
+
+  private QueryPageMock readQueryPageMock(String name) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    String template = String.format("fi/metatavu/edelphi/test/mock/%s.json", name);
+
+    try (InputStream templateStream = getClass().getClassLoader().getResourceAsStream(template)) {
+      return objectMapper.readValue(templateStream, QueryPageMock.class);
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, String.format("Failed to read template query page %s", name), e);
+      fail(String.format("Failed to read template query page %s", template));
+    }
+
+    return null;
+  }
+
+  private long createQuery(String name, String urlName, Long parentFolderId, boolean visible, Long creatorId,
+      Date created, Long modifierId, Date modified, String description, Integer indexNumber, boolean allowEditReply,
+      Date closes, String state) {
+    long id = getNextId("Resource");
+    createQuery(id, name, urlName, parentFolderId, visible, creatorId, created, modifierId, modified, description,
+        indexNumber, allowEditReply, closes, state);
+    queryIds.add(id);
+    return id;
+  }
+
+  private void createQuery(Long id, String name, String urlName, Long parentFolderId, boolean visible, Long creatorId,
+      Date created, Long modifierId, Date modified, String description, Integer indexNumber, boolean allowEditReply,
+      Date closes, String state) {
+    String resourceSql = "INSERT INTO "
+        + "  `resource` (`id`, `name`, `urlName`, `parentFolder_id`, `type`, `visible`, `archived`, `creator_id`, `created`, `lastModifier_id`, `lastModified`, `description`, `indexNumber`) "
+        + "VALUES " + "  (?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+    String querySql = "INSERT INTO " + "  `query` (`id`, `allowEditReply`, `closes`, `state`) " + "VALUES "
+        + "  (?,?,?,?)";
+
+    executeSql(resourceSql, id, name, urlName, parentFolderId, "QUERY", visible, false, creatorId, created, modifierId,
+        modified, description, indexNumber);
+    executeSql(querySql, id, allowEditReply, closes, state);
+  }
+
+  private Long createQuerySection(Long queryId, String title, Long creatorId, Date created, Long modifierId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)